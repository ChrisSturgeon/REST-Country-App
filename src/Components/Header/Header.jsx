import React, { useContext } from 'react';
import { StyledHeader } from './Header.style';

export default function Header({ toggleDarkMode }) {
  // const { darkMode, toggleDarkMode } = useContext(DarkModeContext);

  return (
    <StyledHeader>
      <h1>Where in the World?</h1>
      {/* <button
        onClick={() => {
          console.log(darkMode);
          toggleDarkMode();
        }}
      >
        Hi
      </button> */}
      <button onClick={toggleDarkMode}>Change Theme</button>
    </StyledHeader>
  );
}
