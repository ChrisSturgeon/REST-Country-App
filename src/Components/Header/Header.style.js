import styled from 'styled-components';
import { device } from '../../../devices';

export const StyledHeader = styled.header`
  display: flex;
  justify-content: space-between;
  background-color: ${(props) => props.theme.primary};
  color: ${(props) => props.theme.text};

  @media ${device.tablet} {
    padding: 2em;
  }
`;
