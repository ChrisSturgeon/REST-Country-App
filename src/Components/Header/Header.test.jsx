import { render, screen } from '@testing-library/react';
import React from 'react';
import Header from './Header';

describe('Header Tests', () => {
  it.skip('Matches snapshot', () => {
    render(<Header />);
    expect(document.body).toMatchSnapshot();
  });

  it('Has "header" element as banner', () => {
    render(<Header />);
    expect(screen.getByRole('banner')).toBeInTheDocument();
  });

  it('Should have a header element', () => {
    render(<Header />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });

  it('Should render h1 level heading', () => {
    render(<Header />);
    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
  });

  it('Has correct title', () => {
    render(<Header />);
    expect(screen.getByRole('heading')).toHaveTextContent(
      'Where in the World?'
    );
  });
});
