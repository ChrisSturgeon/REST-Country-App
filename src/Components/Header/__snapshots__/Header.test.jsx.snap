// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Matches snapshot 1`] = `
<body>
  <div>
    <h1>
      Hoi
    </h1>
  </div>
</body>
`;
